New Entry: PHP array_rand() (#2956)
Codecademy/docs: content/php/concepts/arrays/terms/array-rand/array-rand.md

---

> Title: array_rand()

Reviewer: ```suggestion
Title: 'array_rand()'
```

> Description: Used to select one or more random keys from an array.

Reviewer: ```suggestion
Description: 'Used to select one or more random keys from an array.'
```

Subjects:
  - 'Computer Science'
  - 'Web Development'
Tags:
  - 'Arrays'
  - 'Methods'
  - 'Functions'
CatalogContent:
  - 'learn-php'
  - 'paths/computer-science'
---

> The **array_rand()** function returns either:
> - a random key from an array, or
> - an array of random keys from an array.

Reviewer: ```suggestion
The **`array_rand()`** function returns either a random key from an array, or an array of random keys from an array.
```

## Syntax

```pseudo
array_rand($array, $number = 1)
```

- `$array`: The array from which the keys will be randomly selected.
- `$number` (optional): Specifies the number of random keys to be selected from the array. By default, it is set to 1.

> **Note:** When trying to pick more elements than there are in the array, `array_rand()` will return `NULL` and throw an `E_WARNING` error.

## Example

Consider the following array of fruits.

```php
<?php
  $fruits = array("apple", "banana", "orange", "grape", "mango");
?>
```

> To get a single key from the array, 

Reviewer: ```suggestion
To get a single key from the array:
```

```php
<?php
  $randomKey = array_rand($fruits);
  print_r($randomKey)
?>
```

The above code might result in the following output:

```shell
2
```

> To get three random keys from the array,

Reviewer: ```suggestion
To get three random keys from the array:
```
